core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/version/VersionCreateRequest.java

```diff
@@ -195,10 +197,35 @@ public Boolean execute(RepositoryContext context) {
 				monitor.worked(1);
 //			});
 			
+			//Find resources that depend on this resource
+			List<String> dependentResourceIds = ResourceRequests.prepareSearch()
+				.setLimit(10_000)
+				.filterByDependency(String.format("(uri:%s OR uri:%s/* OR uri:%s\\?*) AND scope:domain) ", resource, resource, resource))
+				.stream(context)
+				.flatMap(Resources::stream)
+				.map(Resource::getId)
+				.collect(Collectors.toList());
+			
+			if (resourceToVersion instanceof TerminologyResourceCollection) {
```

Comment:
This is incorrect, when versioning a collection the system will version all child resources. This will be handled in a different issue, please remove this block.